src/search/pdbs/validation.cc

```diff
@@ -40,18 +41,19 @@ void validate_and_normalize_pattern(const TaskProxy &task_proxy,
 }
 
 void validate_and_normalize_patterns(const TaskProxy &task_proxy,
-                                     PatternCollection &patterns) {
+                                     PatternCollection &patterns,
+                                     utils::LogProxy &log) {
     /*
       - Validate and normalize each pattern (see there).
       - Warn if duplicate patterns exist.
     */
     for (Pattern &pattern : patterns)
-        validate_and_normalize_pattern(task_proxy, pattern);
+        validate_and_normalize_pattern(task_proxy, pattern, log);
     PatternCollection sorted_patterns(patterns);
     sort(sorted_patterns.begin(), sorted_patterns.end());
     auto it = unique(sorted_patterns.begin(), sorted_patterns.end());
     if (it != sorted_patterns.end()) {
-        utils::g_log << "Warning: duplicate patterns have been detected" << endl;
+        log << "Warning: duplicate patterns have been detected" << endl;
```

Comment:
Use log-level for warnings and to be more consistent over the entire codebase write WARNING in all caps (or adapt everywhere else).